python/vyos/ifconfig/vxlan.py

```diff
@@ -66,24 +66,26 @@ class VXLANIf(Interface):
         'ifname': 'add',
         'vni':    'id',
         'port':   'dstport',
-        'src_address': 'nolearning local',
+        'src_address': 'local',
+        'src_interface': 'dev',
     }
 
     def _create(self):
-        cmdline = set()
-        if self.config['remote']:
-            cmdline = ('ifname', 'type', 'remote', 'src_interface', 'vni', 'port')
-
-        elif self.config['src_address']:
-            cmdline = ('ifname', 'type', 'src_address', 'vni', 'port')
+        cmdline = ['ifname', 'type', 'vni', 'port'] 
 
-        elif self.config['group'] and self.config['src_interface']:
-            cmdline = ('ifname', 'type', 'group', 'src_interface', 'vni', 'port')
+        if self.config['src_address']:
+            cmdline.append('src_address')
 
-        else:
-            ifname = self.config['ifname']
-            raise ConfigError(
-                f'VXLAN "{ifname}" is missing mandatory underlay interface for a multicast network.')
+        if self.config['remote']:
+            cmdline.append('remote')
+
+        if self.config['group'] or self.config['src_interface']:
```

Comment:
Could this be better written as ?

if self.config['group'] and self.config['src_interface']:
    cmdline.append('group', 'src_interface')
elif self.config['group'] or self.config['src_interface']:
     ifname = self.config['ifname']
     raise ConfigError(
                     f'VXLAN "{ifname}" is missing mandatory underlay multicast group or source interface for a multicast network.')